receiver/sqlqueryreceiver/logs_receiver.go

```diff
@@ -0,0 +1,318 @@
+	}
+	close(receiver.shutdownRequested)
+}
+
+type logsQueryReceiver struct {
+	id           string
+	query        Query
+	createDb     dbProviderFunc
+	createClient clientProviderFunc
+	logger       *zap.Logger
+
+	db            *sql.DB
+	client        dbClient
+	trackingValue string
+	// TODO: Extract persistence into its own component
+	storageClient           storage.Client
+	trackingValueStorageKey string
+}
+
+func newLogsQueryReceiver(
+	id string,
+	query Query,
+	dbProviderFunc dbProviderFunc,
+	clientProviderFunc clientProviderFunc,
+	logger *zap.Logger,
+	storageClient storage.Client,
+) *logsQueryReceiver {
+	queryReceiver := &logsQueryReceiver{
+		id:            id,
+		query:         query,
+		createDb:      dbProviderFunc,
+		createClient:  clientProviderFunc,
+		logger:        logger,
+		storageClient: storageClient,
+	}
+	queryReceiver.trackingValue = queryReceiver.query.TrackingStartValue
+	queryReceiver.trackingValueStorageKey = fmt.Sprintf("%s.%s", queryReceiver.id, "trackingValue")
+	return queryReceiver
+}
+
+func (queryReceiver *logsQueryReceiver) ID() string {
+	return queryReceiver.id
+}
+
+func (queryReceiver *logsQueryReceiver) start(ctx context.Context) error {
+	var err error
+	queryReceiver.db, err = queryReceiver.createDb()
+	if err != nil {
+		return fmt.Errorf("failed to open db connection: %w", err)
+	}
+	queryReceiver.client = queryReceiver.createClient(dbWrapper{queryReceiver.db}, queryReceiver.query.SQL, queryReceiver.logger)
+
+	queryReceiver.trackingValue = queryReceiver.retrieveTrackingValue(ctx)
+
+	return nil
+}
+
+// retrieveTrackingValue retrieves the tracking value from storage, if storage is configured.
+// Otherwise, it returns the tracking value configured in `tracking_start_value`.
+func (queryReceiver *logsQueryReceiver) retrieveTrackingValue(ctx context.Context) string {
+	trackingValueFromConfig := queryReceiver.query.TrackingStartValue
+	if queryReceiver.storageClient == nil {
+		return trackingValueFromConfig
+	}
+
+	storedTrackingValueBytes, err := queryReceiver.storageClient.Get(ctx, queryReceiver.trackingValueStorageKey)
+	if err != nil || storedTrackingValueBytes == nil {
+		return trackingValueFromConfig
+	}
+
+	return string(storedTrackingValueBytes)
+
+}
+
+func (queryReceiver *logsQueryReceiver) collect(ctx context.Context) (plog.Logs, error) {
+	logs := plog.NewLogs()
+
+	var rows []stringMap
+	var err error
+	if queryReceiver.query.TrackingColumn != "" {
+		rows, err = queryReceiver.client.queryRows(ctx, queryReceiver.trackingValue)
+	} else {
+		rows, err = queryReceiver.client.queryRows(ctx)
+	}
+	if err != nil {
+		return logs, fmt.Errorf("error getting rows: %w", err)
+	}
+
+	var errs error
+	scopeLogs := logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords()
+	for logsConfigIndex, logsConfig := range queryReceiver.query.Logs {
+		for i, row := range rows {
+			if err = rowToLog(row, logsConfig, scopeLogs.AppendEmpty()); err != nil {
```

Comment:
`rowToLog` doesn't return any error. Let's clean up the redundant error handling?

Reply:
OK, removed. I had left the error checking in there because I expected the `rowToLog` function to return errors in the future, but yeah, we can add error checking when it's needed.